.ci/go-test.sh

```diff
@@ -240,8 +240,11 @@ main()
 
 	[ -z "$test_packages" ] && echo "INFO: no golang code to test" && exit 0
 
+	local go_ldflags
+	[ "$(uname -m)" = s390x ] && go_ldflags="-extldflags -Wl,--s390-pgste"
```

Comment:
Nit: on https://github.com/kata-containers/tests/tree/main/.ci/go-test.sh#L26 it uses `"$(go env GOARCH)"` to get the arch. In case you want to keep it consistent.

Yet-another-Nit: do you need to pass `-Wl`? If I am not wrong, in C, `-Wl,` is used to pass a linker argument via compiler. Given that you are passing it direct to the linker via `-extldflags` maybe you don't need it. Not a golang expert speaking...

Reply:
> Nit: on https://github.com/kata-containers/tests/tree/main/.ci/go-test.sh#L26 it uses `"$(go env GOARCH)"` to get the arch. In case you want to keep it consistent.

Consistency is another can of worms :) but okay, let me update this one

> Yet-another-Nit: do you need to pass `-Wl`? If I am not wrong, in C, `-Wl,` is used to pass a linker argument via compiler. Given that you are passing it direct to the linker via `-extldflags` maybe you don't need it. Not a golang expert speaking...

`-extldflags` is, slightly counterintuitively, passed to GCC: `s390x-linux-gnu-gcc: error: unrecognized command line option ‘--s390-pgste’`

Reply:
> > Nit: on https://github.com/kata-containers/tests/tree/main/.ci/go-test.sh#L26 it uses `"$(go env GOARCH)"` to get the arch. In case you want to keep it consistent.
> 
> Consistency is another can of worms :) but okay, let me update this one
> 

hehehe unfortunately it is true

> > Yet-another-Nit: do you need to pass `-Wl`? If I am not wrong, in C, `-Wl,` is used to pass a linker argument via compiler. Given that you are passing it direct to the linker via `-extldflags` maybe you don't need it. Not a golang expert speaking...
> 
> `-extldflags` is, slightly counterintuitively, passed to GCC: `s390x-linux-gnu-gcc: error: unrecognized command line option ‘--s390-pgste’`

hmm... interesting. Thanks for checking that.